lib/features/email/presentation/widgets/message_content_tile_builder.dart

```diff
@@ -57,47 +51,21 @@ class MessageContentTileBuilder {
     final message = (_attachmentInlines.isNotEmpty && _session != null && _accountId != null && _messageContent.hasImageInlineWithCid())
       ? '${_messageContent.getContentHasInlineAttachment(_session!, _accountId!, _attachmentInlines)}'
       : '${_messageContent.content}';
-    return message;
-  }
-
-  String cleanHtmlMessage(String message) => hasTagTable(getHtmlMessageText()) ? sanitizeHtml(getHtmlMessageText()) : message;
-
-  bool hasTagTable(String html) => html.toLowerCase().contains('</table>');
-}
-
-class _PopupPhotoViewWidgetFactory extends WidgetFactory {
 
-  final ImagePaths _imagePaths;
+    final trustAsHtml = purifyHtmlMessage(message);
 
-  _PopupPhotoViewWidgetFactory(this._imagePaths);
-
-  @override
-  Widget? buildImageWidget(BuildMetadata meta, {String? semanticLabel, required String url}) {
-    final built = super.buildImageWidget(meta, semanticLabel: semanticLabel, url: url);
-
-    if (built is Image) {
-      return Builder(
-        builder: (context) => GestureDetector(
-          onTap: () => Navigator.of(context).push(MaterialPageRoute(
-              builder: (_) => Scaffold(
-                body: Stack(
-                  children: [
-                    PhotoView(
-                      heroAttributes: PhotoViewHeroAttributes(tag: url),
-                      imageProvider: built.image),
-                    Padding(
-                      padding: EdgeInsets.only(left: 16, top: 35),
-                      child: IconButton(
-                        onPressed: () => Navigator.of(context).pop(),
-                        icon: SvgPicture.asset(_imagePaths.icCloseMailbox, width: 30, height: 30, fit: BoxFit.fill)))
-                  ],
-                ),
-              ))),
-          child: Hero(tag: url, child: built),
-        ),
-      );
-    }
+    return trustAsHtml;
+  }
 
-    return built;
+  String purifyHtmlMessage(String html) {
+    return sanitizeHtml(
+        html,
+        allowAttributes: (attribute) => attribute == 'style' || attribute == 'input' || attribute == 'form')
+      .changeStyleBackground()
+      .removeFontSizeZeroPixel()
+      .removeMaxHeightZeroPixel()
+      .removeMaxWidthZeroPixel()
+      .removeHeightZeroPixel()
+      .removeWidthZeroPixel();
```

Comment:
please split to another class for `Purifying`, some other place can re-use it. `HtmlMessagePurifier`